Esapi version updated

MarekB01 opened the pull request:
https://github.com/BroadleafCommerce/QA/issues/4694
upgraded org.owasp.esapi to version 2.3.0.0 (latest version)

MarekB01 committed:
BroadleafCommerce/QA#4694

upgraded org.owasp.esapi to version 2.3.0.0 (latest version)

MarekB01 committed:
BroadleafCommerce/QA-4694

Made some changes to POM file

riteshadhikari17 reviewed (changes requested):
When you upgrade lib, you have to verify that the version is compatible. For that you have to run tests. Always run `mvn clean install` without skipping tests before opening any PR. For the big libs like spring, hibernate, esapi, jquery etc. You should do additional verification by running ReferenceSite.
There are test failures so look at 6.2 and see what was done when upgrading esapi.

MarekB01 committed:
BroadleafCommerce/QA-4694
Runned mvn clea install with tests, also tested this on Admin application and Site, everything works good

MarekB01 committed:
BroadleafCommerce/QA-4694
Added esapi-java-logging.properties file
Runned mvn clea install with tests, also tested this on Admin application and Site, everything works good
